torch/_torch_docs.py

```diff
@@ -3508,25 +3508,164 @@ def merge_dicts(*dicts):
+
+From Taylor's theorem for a function :math:`f \in C^3` (:math:`f` has at least 3
+continuous derivatives) we have:
+
+.. math::
+    \begin{aligned}
+        f(x+h_r ) &= f(x) + h_r f'(x) + {h_r}^2  \frac{f''(x)}{2} + {h_r}^3 \frac{f'''(x_r)}{6} \\
+        f(x-h_l ) &= f(x) - h_l f'(x) + {h_l}^2 \frac{f''(x)}{2} - {h_l}^3 \frac{f'''(x_l)}{6}  \\
+        f(x ) &= f(x)
+    \end{aligned}
+
+where :math:`x_r` and :math:`x_l` are two numbers in intervals :math:`[x, x+ h_r]` and
+:math:`[x-h_l, x]` correspondingly. To derive :math:`f'(x)` from equations above, we multiply the
+first equation by :math:`{h_l}^2`, the second equation by :math:`-{h_r}^2` and the third one by
+:math:`{h_r}^2-{h_l}^2` and sum of these expressions gives us the following:
+
+.. math::
+    \begin{aligned}
+        {h_l}^2 f(x+h_r) - &{h_r}^2 f(x-h_l) + ({h_r}^2-{h_l}^2 ) f(x) = \\
+        &{h_l}^2 {h_r} f'(x) + {h_l}^2 {h_r}^3 \frac{f'''(x_r)}{6} + {h_r}^2
+        {h_l} f'(x) + {h_r}^2 {h_l}^3 \frac{f'''(x_l)}{6}
+    \end{aligned}
+
+
+after simple algebraic manipulations we get:
+
+.. math::
+    \begin{aligned}
+        {h_r} {h_l}^2 f'(x) + {h_r}^2 {h_l} f'(x) = &{h_l}^2 f(x+h_r) -
+            {h_r}^2 f(x-h_l) + ({h_r}^2-{h_l}^2 ) f(x) \\
+        &- {h_l}^2 {h_r}^3 \frac{f'''(x)}{6} + {h_r}^2 {h_l}^3
+            \frac{f'''(x)}{6}
+    \end{aligned}
+
+which concludes:
+
+.. math::
+    f'(x) =  \frac{ {h_l}^2 f(x+h_r) -{h_r}^2 f(x-h_l)
+            + ({h_r}^2-{h_l}^2 ) f(x) }{ {h_r} {h_l}^2 + {h_r}^2 {h_l} }
+            - {h_r}{h_l} \frac{ (f'''(x_r) + f'''(x_l))}{6}
+
+Because continuous functions are bounded in a closed domain and :math:`f \in C^3` , we can conclude
+when the multiplication :math:`h_r h_l \rightarrow 0` then we have:
+
+.. math::
+    f'(x) =  \frac{ {h_l}^2 f(x+h_r) -{h_r}^2 f(x-h_l)
+             + ({h_r}^2-{h_l}^2 ) f(x) }{ {h_r} {h_l}^2 + {h_r}^2 {h_l} }
+             - \mathcal{O}({h_r}{h_l}).
+
+
+Therefore, we can estimate :math:`f'(x)` by
+
+.. math::
+    f'(x) \approx  \frac{ {h_l}^2 f(x+h_r) -{h_r}^2 f(x-h_l)
+          + ({h_r}^2-{h_l}^2 ) f(x) }{ {h_r} {h_l}^2 + {h_r}^2 {h_l} }
+
+An interesting special case, happens when :math:`h_r = h_l=h` which gives us
+:math:`f'(x) \approx  \frac{  f(x+h) - f(x-h) }{ 2h}`.
+
+.. note::
+    We estimate the gradient of functions in complex domain
+    :math:`g : \mathbb{C}^n \rightarrow \mathbb{C}` in the same way.
 
 Args:
-    {input}
+    input (``Tensor``): the tensor that represents the values of the function
 
 Keyword args:
-    spacing (scalar, list of scalar, list of Tensor, optional): implicitly or explicitly represents
-    the coordinates the function is evaluated at
-    dim (int, list of int, optional): the dimension or dimensions to approximate the gradient over.
-    edge_order (int, optional): unsupported (must be equal to its default value which is 1.)
+    spacing (``scalar``, ``list of scalar``, ``list of Tensor``, optional): i) list of 1-dimensional Tensors
```

Comment:
Start the description of spacing by reminding the reader about the earlier description: 

:attr:\`spacing\` can be used to modify how the :attr:\`input\` tensor's indices relate to sample coordinates. If :attr:\`spacing\` is a scalar then the indices are multiplied by the scalar to produce the coordinates. For example, if `spacing=2` the indices (1, 2, 3) become coordinates (2, 4, 6). If :attr:\spacing\` is a list of scalars then the corresponding indices are multiplied. For example, if `spacing=(2, -1, 3)` the indices (1, 2, 3) become coordinates (2, -2, 9). Finally, if :attr:\`spacing\` is a list of one-dimensional tensors then each tensor specifies the coordinates for the corresponding dimension. For example, if the indices are (1, 2, 3) and the tensors are (t0, t1, t2), then the coordinates are (t0[1], t1[2], t2[3]). 

What would you think of a description like that?

Reply:
this sounds great ! thanks for suggestion.